File: televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/viewmodel/StreamingSourceViewModelItem.java
package com.vanillax.televisionbingecalculator.app.viewmodel;

import com.vanillax.televisionbingecalculator.app.R;

/**
 * Created by mitchross on 3/5/17.
 */

public class StreamingSourceViewModelItem
{
	public static final String NETFLIX = "netflix";
	public static final String HULU = "hulu";
	public static final String AMAZON = "amazon";
	public static final String VUDU = "vudu";
	public static final String HBO = "hbo";
	String streamSourceMovie;
	


	public StreamingSourceViewModelItem( String streamSource )
	{
		this.streamSourceMovie = streamSource;
	}


	public int getStreamingIcon()
	{
		String streamingIconName = "";
//		if( streamSource !=null)
//		{
//			streamingIconName = streamSource.sourceDisplayName.toLowerCase();
//		}
		 if ( streamSourceMovie !=null || !streamSourceMovie.isEmpty())
		{
			streamingIconName = streamSourceMovie.toLowerCase();
		}
		
			
		

		if( streamingIconName.contains( NETFLIX ))
        {

			return  R.drawable.netflix;
        }
        if ( streamingIconName.contains( HULU ))
        {
			return R.drawable.hulu;
        }
        if ( streamingIconName.contains( AMAZON ))
        {
			return R.drawable.amazon;
        }
        if ( streamingIconName.contains( VUDU ))
        {
			return R.drawable.vudu;
        }
		if ( streamingIconName.contains( HBO ))
		{
			return R.drawable.hbo;
		}
        else
		{
			return 0;
		}

	}
}

File: televisionbingecalculator/src/main/java/com/vanillax/televisionbingecalculator/app/viewmodel/StreamingSourceViewModelItem.kt
package com.vanillax.televisionbingecalculator.app.viewmodel

import androidx.annotation.DrawableRes
import com.vanillax.televisionbingecalculator.app.R
import com.vanillax.televisionbingecalculator.app.kotlin.items.BaseItem

/**
 * Created by mitchross on 3/5/17.
 */

class StreamingSourceViewModelItem(private var videoStream: VideoStream?): BaseItem(R.layout.streaming_source) {


    //		if( streamSource !=null)
    //		{
    //			streamingIconName = streamSource.sourceDisplayName.toLowerCase();
    //		}
    val streamingIcon: Int
        get() = videoStream?.icon ?: 0

    enum class VideoStream(@DrawableRes val icon: Int) {
        NETFLIX(R.drawable.netflix),
        HULU(R.drawable.hulu),
        AMAZON(R.drawable.amazon),
        VUDO(R.drawable.vudu),
        HBO(R.drawable.hbo)
    }
}
